src/native/libs/System.Globalization.Native/pal_collation.c

```diff
@@ -321,7 +321,24 @@ static UCollator* CloneCollatorWithOptions(const UCollator* pCollator, int32_t o
 
     if (customRuleLength == 0)
     {
+#if !defined(STATIC_ICU)
+        if (ucol_clone_ptr != NULL)
+        {
+            pClonedCollator = ucol_clone(pCollator, pErr);
```

Comment:
No, `ucol_clone` work as all other ICU APIs we use. It is not different. I am using `ucol_clone_ptr` for the null check because I cannot use `ucol_clone` for that check.

Reply:
Hmm, I am not sure how this would link in case the `ucol_clone` doesn't exist. While it would not be ever called in that case, the linker would still need to resolve the symbol, right? Doesn't it leave the libSystem.Globalization.Native.so with undefined reference to `ucol_clone` in that case?

Reply:
The examples I gave is for `ucol_setMaxVariable` but this apply the same way for `ucol_clone`

Reply:
> so with undefined reference to ucol_clone in that case?

That is right, therefore we always check for `_ptr` before calling this method. We never call it if we couldn't find the symbols.

Reply:
Sigh, time has eroded my memory on these and I've forgotten that the `ucol_clone` in this case is the macro that is defined to call via the related `_ptr`. I am sorry for the confusion, then everything is good.